conn2res/plotting.py

```diff
@@ -169,41 +179,43 @@ def plot_iodata(
     y = _check_y_dims(y)
 
     # set plotting theme
-    sns.set(style="ticks", font_scale=1.0)
-    fig, ax = plt.subplots(1, 1, figsize=(12, 2))  # 12, 4.5
-
-    # set color palette
-    palette = kwargs.pop('palette', None)
+    rc_defaults = {'figure.titlesize': 12, 'axes.labelsize': 11,
+                   'xtick.labelsize': 11, 'ytick.labelsize': 11,
+                   'legend.fontsize': 8, 'legend.loc': 'best',
+                   'lines.linewidth': 1, 'savefig.format': 'png'}
+    rc_defaults.update(rc_params)
+    sns.set_theme(style='ticks', rc=rc_defaults)
+    
+    # open figure and axes
+    fig_defaults = {'figsize': (12, 2)}  # 12, 4.5
+    fig_defaults.update(fig_params)
+    fig = plt.figure(**fig_defaults)
+    ax = fig.subplots(1, 1)
 
     # plot inputs (x) and outputs (y)
     sns.lineplot(
         data=x, palette=palette, dashes=False, legend=False, ax=ax,
-        linewidth=1.0, **kwargs
+        **kwargs
     )
     sns.lineplot(
         data=y, palette=palette, dashes=False, legend=False, ax=ax,
         linewidth=1.5, **kwargs
     )
-    ax.set_xlim(0, 200)
-
-    # set axis labels
-    ax.set_xlabel('time steps', fontsize=11)
-    ax.set_ylabel('signal amplitude', fontsize=11)
-
-    # xlabels, ylabels
-    if x.ndim == 1:
-        x_labels = ['x']
-    else:
-        x_labels = [f'x{n+1}' for n in range(x.shape[1])]
-    if y.ndim == 1:
-        y_labels = ['y']
-    else:
-        y_labels = [f'y{n+1}' for n in range(y.shape[1])]
 
     # set legend
-    new_labels = x_labels + y_labels
-    ax.legend(handles=ax.lines, labels=new_labels, loc='best',
-              fontsize=8)
+    x_labels = ['x'] if x.ndim == 1 else [f'x{n+1}' for n in range(x.shape[1])]
+    y_labels = ['y'] if y.ndim == 1 else [f'y{n+1}' for n in range(y.shape[1])]
+    lg_defaults = {'labels': x_labels + y_labels}
+    lg_defaults.update(**lg_params)
+    ax.legend(handles=ax.lines, **lg_defaults)
+
+    # set axes properties
+    ax_defaults = {'xlabel': 'time steps', 'ylabel': 'signal amplitude',
+                   'xlim': [0, 200]}
+    if title is not None:
+        ax_defaults.update(title=title)
```

Comment:
I change the title of the figure here for backward compatibility (before plt.title was used which sets title at an axes level)